src/DBFTPlugin/Consensus/ConsensusContext.Get.cs

```diff
@@ -66,12 +78,21 @@ private PreparationPayloadCompact GetPreparationPayloadCompact(ExtensiblePayload
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public byte GetPrimaryIndex(byte viewNumber)
+        public byte GetPriorityPrimaryIndex(byte viewNumber)
         {
-            int p = ((int)Block.Index - viewNumber) % Validators.Length;
+            int p = ((int)Block[0].Index - viewNumber) % Validators.Length;
             return p >= 0 ? (byte)p : (byte)(p + Validators.Length);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public byte GetFallbackPrimaryIndex(byte viewNumber, byte priorityPrimaryIndex)
```

Comment:
viewNumber is not needed for this formula, because fallback is always 0

Reply:
Done